sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientBuilder.java

```diff
@@ -0,0 +1,248 @@
+     * @throws IllegalStateException If {@link CryptographyClientBuilder#credential(TokenCredential)} or
+     * either of ({@link CryptographyClientBuilder#keyIdentifier(String)} or {@link CryptographyClientBuilder#jsonWebKey(JsonWebKey)} have not been set.
+     */
+    public CryptographyClient buildClient() {
+        return new CryptographyClient(buildAsyncClient());
+    }
+    /**
+     * Creates a {@link CryptographyAsyncClient} based on options set in the builder.
+     * Every time {@code buildAsyncClient()} is called, a new instance of {@link CryptographyAsyncClient} is created.
+     *
+     * <p>If {@link CryptographyClientBuilder#pipeline(HttpPipeline) pipeline} is set, then the {@code pipeline} and
+     * ({@link CryptographyClientBuilder#keyIdentifier(String) jsonWebKey identifier} or {@link CryptographyClientBuilder#jsonWebKey(JsonWebKey) json web jsonWebKey})
+     * are used to create the {@link CryptographyAsyncClient async client}. All other builder settings are ignored. If {@code pipeline} is not set,
+     * then ({@link CryptographyClientBuilder#credential(TokenCredential) jsonWebKey vault credential} and
+     * ({@link CryptographyClientBuilder#keyIdentifier(String) jsonWebKey identifier} or {@link CryptographyClientBuilder#jsonWebKey(JsonWebKey) json web jsonWebKey})
+     * are required to build the {@link CryptographyAsyncClient async client}.</p>
+     *
+     * @return A {@link CryptographyAsyncClient} with the options set from the builder.
+     * @throws IllegalStateException If {@link CryptographyClientBuilder#credential(TokenCredential)} or
+     * either of ({@link CryptographyClientBuilder#keyIdentifier(String)} or {@link CryptographyClientBuilder#jsonWebKey(JsonWebKey)} have not been set.
+     */
+    public CryptographyAsyncClient buildAsyncClient() {
+        Configuration buildConfiguration = (configuration == null) ? ConfigurationManager.getConfiguration().clone() : configuration;
+
+        if (jsonWebKey == null && Strings.isNullOrEmpty(keyId)) {
+            throw new IllegalStateException("Json Web Key or jsonWebKey identifier are required to create cryptography client");
+        }
+
+        if (pipeline != null) {
+            if(jsonWebKey != null) {
+                return new CryptographyAsyncClient(jsonWebKey, pipeline);
+            } else {
+                return new CryptographyAsyncClient(keyId, pipeline);
+            }
+        }
+
+        if (credential == null) {
+            throw new IllegalStateException(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.CREDENTIAL_REQUIRED));
+        }
+
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+        policies.add(new UserAgentPolicy(AzureKeyVaultConfiguration.SDK_NAME, AzureKeyVaultConfiguration.SDK_VERSION, buildConfiguration));
+        policies.add(retryPolicy);
```

Comment:
Need to add `addBeforeRetryPolicies()` and `addAfterRetryPolicies()` for tracing implementation to be included in th pipeline.

Reply:
added